macros/audit.sql

```diff
@@ -1,37 +1,54 @@
+{% macro get_audit_schema() %}
+
+    {# if the get_audit_schema macro exists in the base project use that #}
+    {% if context.get(project_name, {}).get('get_audit_schema') %}
+        {{ return(context[project_name].get_audit_schema()) }}
+    {% else %}
+        {{ return(target.schema~'_meta') }}
+    {% endif %}
+
+{% endmacro %}
+
 {% macro get_audit_relation() %}
+    {%- set audit_schema=logging.get_audit_schema() -%}
+
     {%- set audit_table =
         api.Relation.create(
+            database=target.database,
+            schema=audit_schema,
             identifier='dbt_audit_log',
-            schema='dbt_meta',
             type='table'
         ) -%}
-    {{ return(audit_table) }}
-{% endmacro %}
 
+    {{ return(audit_table) }}
 
-{% macro get_audit_schema() %}
-    {% set audit_table = logging.get_audit_relation() %}
-    {{ return(audit_table.include(schema=True, identifier=False)) }}
 {% endmacro %}
 
-
-{% macro log_audit_event(event_name, schema, relation) %}
+{% macro log_audit_event(event_name, schema, relation, user, target_name, is_full_refresh) %}
 
     insert into {{ logging.get_audit_relation() }} (
         event_name,
         event_timestamp,
         event_schema,
         event_model,
+        event_user,
+        event_target,
+        event_is_full_refresh,
         invocation_id
-        )
+    )
 
     values (
         '{{ event_name }}',
-        convert_timezone('UTC', current_timestamp)::timestamp_ntz,
-        {% if variable != None %}'{{ schema }}'{% else %}null::varchar(512){% endif %},
-        {% if variable != None %}'{{ relation }}'{% else %}null::varchar(512){% endif %},
+        {{ dbt_utils.current_timestamp_in_utc() }},
```

Comment:
@bobzoller so this is what we actually just got rid of as per Snowflake's recommendation (redundant casting)

Reply:
ah k